frontend/individual_pedigree.tsx

```diff
@@ -0,0 +1,160 @@
+/**
+ * @file This file contains the Pedigree function. This function fetches
+ *       pedigree for a given `id` (parsed from the url).
+ */
+
+import React, { Component, createRef } from 'react'
+import { Link } from "react-router-dom";
+import { get } from './communication';
+import { Network } from 'vis-network';
+import "vis-network/styles/vis-network.css"
+
+
+
+/**
+ * Shows the information of a given individual and the pedigree graph built using the vis-network component
+ */
+export function IndividualPedigree({ id, generations }: { id: string }) {
+  const [pedigree, setPedigree] = React.useState(undefined as any)
+  const [individual, setIndividual] = React.useState(undefined as any)
+  const [generations_input, setGenerations] = React.useState(generations)
+
+  React.useEffect(() => {
+    get(`/api/individual/${id}`).then(
+      data => data && setIndividual(data),
+      error => console.error(error)
+    )
+    get(`/api/pedigree/${id}/${generations_input}`).then(
+      data => data && setPedigree(data),
+      error => console.error(error)
+    )
+  }, [id, generations_input])
+
+  class PedigreeNetwork extends Component {
+
+    options = {
+      width: Math.round(window.innerWidth * 0.90) + 'px',
+      height: Math.round(window.innerHeight * 0.80) + 'px',
+      layout: {
+        hierarchical: {
+          enabled: true,
+          direction: "DU",
+          sortMethod: 'directed',
+          //shakeTowards: "roots",
+          levelSeparation: 150,
+          parentCentralization:true,
+          edgeMinimization: true,
+          blockShifting: true,
+          nodeSpacing: 100
+        }
+      },
+      edges: {
+        /*color:  { color: "gray", inherit: false },*/
+        arrows: { to: { enabled: true, scaleFactor: 0.50 } }
+        //smooth: { type: "cubicBezier", forceDirection: "vertical", roundness: 1 }
+      },
+      interaction: {
+        navigationButtons: true,
+        keyboard: true
+      },
+      physics: false,
+      /* physics: {
+        hierarchicalRepulsion: {
+          centralGravity: 0,
+          avoidOverlap: 1,
+          nodeDistance: 150
+        },
+
+        minVelocity: 0.75,
+        solver: "hierarchicalRepulsion"
+      } */
+    }
+
+    constructor() {
+      super();
+      this.network = {};
+      this.appRef = createRef();
+    }
+
+    componentDidMount() {
+      this.network = new Network(this.appRef.current, pedigree, this.options);
+      this.network.on("doubleClick", this.onNodeClick)
+    }
+
+    onNodeClick(params) {
+      if (params.nodes.length > 0) {
+        var nodeid = params.nodes[0];
+        window.location.replace("/pedigree/" + nodeid + "/" + generations_input);
+      }
+    }
+
+    render() {
+
+      return (
+        <div ref={this.appRef}  >
+        </div>
+      );
+    }
+  }
+
+  return <>
+    {individual && pedigree && <>
+      <table width="100%">
```

Comment:
This table is copied from some other file, isn't it?

Reply:
what do yo mean?

Reply:
I used the individual.tsx file as a starting point and added the pedigree graph.

Reply:
should I remove the previous individual.tsx?

Reply:
I don't know, up to you and Martin :)